src/Composer/Satis/PackageSelection/PackageSelection.php

```diff
@@ -126,7 +151,16 @@ public function select(Composer $composer, $verbose)
 
         $repos = $composer->getRepositoryManager()->getRepositories();
         $pool = new Pool($this->minimumStability);
-        foreach ($repos as $repo) {
+        foreach ($repos as $key => $repo) {
+            if ($this->hasRepositoryFilter()) {
+                $repoConfig = $repo->getRepoConfig();
```

Comment:
I think an additional interface with the method 'getRepoConfig()' would be ideal. In the satis code, it's then possible to check if the repository implements this interface. Meanwhile this could be fixed by checking, if the repo extends the ArrayRepository class.